s3/pkg/utils/utils.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+type Database struct {
+	Credential string `conf:"credential,username:password@tcp(ip:port)/dbname"`
+	Driver     string `conf:"driver,mongodb"`
+	Endpoint   string `conf:"endpoint,localhost:27017"`
+}
+
+const (
+	Tier1 = 1
+	Tier99 = 99
+	Tier999 = 999
```

Comment:
Why 99 and 999?  Need some explanations here.

Reply:
1. Tier1, Tier99 and Tier999 just like the tiers of hot, warm, cold.
2. In lifecycle management, objects can not be transitioned from higher tier to lower tier, for example, from Tier99 to Tier1 is not allowed, but from Tier1 to Tier99 is allowed.
3. In the future, we will provide the ability for users to add new storage tiers, if we use 1, 2 and 3, then no space for new storage tiers.
4. The default storage tiers and storage classes are defined as bellow:
	  Default storage classes & tiers:
						T1		            T99				T999
	  AWS S3:				STANDARD	    STANDARD_IA		GLACIER
	  Azure Blob:			HOT			    COOL				ARCHIVE
	  HW OBS:			STANDARD	    WARM				COLD
	  GCP:				Multi-Regional    NearLine			ColdLine
	  Ceph S3:			STANDARD	    -		         		-
	  FusinoStorage Object: 	STANDARD	    -			        	-
	  
          Lifecycle transition:
	  T1   ->  T99:  allowed
	  T1   ->  T999: allowed
	  T99  ->  T999: allowed
      	  T99  ->  T1:   not allowed
	  T999 ->  T1:   not allowed
	  T999 ->  T99:  not allowed

Reply:
Thanks for explanation!  Please add this info in the design doc: https://github.com/opensds/design-specs/pull/24

Reply:
+1 here

Reply:
done!